### DISCUSSION TITLE
Min/max values auto calculation for multiple queries in one panel

### DISCUSSION THREAD

**Discussion by krocans** [grafana/grafana#38132] [Dashboards, Panels & Visualizations]
Looks like min/max "auto" calculation behavior when using multiple queries in single Stat panel has changed in 8.0.6.
In version 8.0.5 min/max values were auto calculated "per query", for example following panel graphs under stat are proportional to each query vales.
![image](https://user-images.githubusercontent.com/42302779/128748529-abbe776e-301b-46c1-ae9b-54b496d1fa93.png)

Since version 8.0.6 (8.1 too) same panel looks following
![image](https://user-images.githubusercontent.com/42302779/128748564-77fd338d-328b-46a2-b5ed-cd08fe104974.png)
It looks like since 8.0.6 common min/max values are calculated per values of all queries together and as a result graphs for series with smaller values becomes almost invisible.

Similar problem is for Bar gauge panel too.

Is this a bug or it's planned change and will remain so in future versions?

**Comment by leeoniya** [comment]
this was a behavior change in 8.x (vs 7.x) that was not entirely intentional. https://github.com/grafana/grafana/issues/32765 illustrates where the new behavior is worse. so we brought back the 7.x logic in https://github.com/grafana/grafana/pull/36497.

i guess the better strategy here would be to have several min/max pairs grouped by units. currently there is just a single min/max that's determined from all numeric fields without regard for units.

**Comment by krocans** [comment]
I see that there are different cases and  in each case need for shared or common minmax is different.
So in my opinion most correct approach will be to make minmax behavior user configurable per panel.

Grouping by units also is not always good. What I see there as example is some heater/heat exchanger intake/supply temperatures being put on one panel - one series data may be about -15C other about +20C an as a result we will get 2 "horizontal lines" one at bottom one at top (or one almost 0 and one full bar in Bar gauge). So it may be useful but also only as configurable option.

**Comment by leeoniya** [comment]
i definitely see and agree with your points. however, we have to balance how many options we expose for the 1% power users but which may overwhelm and/or confuse the remaining 99%.

i think grouping by units is something that is generally better behavior and can be done by default without introducing any additional options, so is most likely to get implemented without the need for much debate on additional options. it also automatically solves the specific example you provided, so there's that ;)

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by nikobearrr** [comment]
I have opened an issue ticket which is the same as I didn't know this was intentionally "fixed".
Would it be a possibility to have it as an option on the Stat panel which min/max to use?

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
